src/futuresboard/cli.py

```diff
@@ -33,7 +33,9 @@ def main():
         help="Disable the routines which scrape while the webservice is running",
     )
     server_settings = parser.add_argument_group("Server Settings")
-    server_settings.add_argument("--host", default=None, help="Server host. Default: 0.0.0.0")
+    server_settings.add_argument(
+        "--host", default='0.0.0.0', help="Server host. Default: 0.0.0.0"
+    )
```

Comment:
```suggestion
    server_settings.add_argument(
        "--host",
        default='0.0.0.0',
        help="Server host. Default: 0.0.0.0",
        type=IPvAnyInterface.validate,
    )
```

Reply:
Don't forget about the import at the top
```diff
+++ b/src/futuresboard/cli.py
@@ -5,6 +5,7 @@ import logging
 import pathlib
 import sys
 
+from pydantic import IPvAnyInterface
 import futuresboard.app
 import futuresboard.scraper
 from futuresboard import __version__  # type: ignore[attr-defined]
```

And leave the `app.run(...)` line untouched.

Reply:
But it does not resolve the issue, does it?
The problem is, that if host argument is NOT defined, than we use the one from config:

    if not args.host:
        args.host = config.HOST

but in this case host argument is defined and thus we don't care about config.HOST (and it's IPvAnyInterface). 
I tried to change it the way you suggest, but still getting the same err:

`Traceback (most recent call last):
  File "/usr/local/bin/futuresboard", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/futuresboard/cli.py", line 61, in main
    app.run(host=args.host.ip.exploded, port=args.port)
AttributeError: 'str' object has no attribute 'ip'`

Reply:
This change, https://github.com/ecoppen/futuresboard/pull/45#discussion_r781142547, will make it properly parse values passed on the CLI

Reply:
Can you please review? It is not working for me: 

`from __future__ import annotations

import argparse
import logging
import pathlib
import sys

from pydantic import IPvAnyInterface
import futuresboard.app
import futuresboard.scraper
from futuresboard import __version__  # type: ignore[attr-defined]
from futuresboard.config import Config

log = logging.getLogger(__name__)


def main():
    parser = argparse.ArgumentParser(prog="futuresboard")
    parser.add_argument("--version", action="version", version=f"%(prog)s {__version__}")
    parser.add_argument(
        "-c",
        "--config-dir",
        type=pathlib.Path,
        default=None,
        help="Path to configuration directory. Defaults to the `config/` sub-directory on the current directory",
    )
    parser.add_argument(
        "--scrape-only", default=False, action="store_true", help="Run only the scraper code"
    )
    parser.add_argument(
        "--disable-auto-scraper",
        default=False,
        action="store_true",
        help="Disable the routines which scrape while the webservice is running",
    )
    server_settings = parser.add_argument_group("Server Settings")
    server_settings.add_argument(
        "--host",
        default='0.0.0.0',
        help="Server host. Default: 0.0.0.0",
        type=IPvAnyInterface.validate,
    )
    server_settings.add_argument(
        "--port", type=int, default=None, help="Server port. Default: 5000"
    )
    args = parser.parse_args()

    if args.config_dir is None:
        args.config_dir = pathlib.Path.cwd() / "config"
    else:
        args.config_dir = args.config_dir.resolve()

    config = Config.from_config_dir(args.config_dir)
    if not args.host:
        args.host = config.HOST
    if not args.port:
        args.port = config.PORT

    # Run the application
    app = futuresboard.app.init_app(config)

    if args.scrape_only:
        with app.app_context():
            futuresboard.scraper.scrape()
        sys.exit(0)

    app.run(host=args.host.ip.exploded, port=args.port)
`


`[root@traders5 futuresboard]# /usr/local/bin/futuresboard --host=10.27.75.50
Traceback (most recent call last):
  File "/usr/local/bin/futuresboard", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/futuresboard/cli.py", line 61, in main
    app.run(host=args.host.ip.exploded, port=args.port)
AttributeError: 'str' object has no attribute 'ip'
`